shell/platform/darwin/ios/platform_message_handler_ios.mm

```diff
@@ -37,7 +34,7 @@ - (void)dispatch:(dispatch_block_t)block {
 namespace flutter {
 
 NSObject<FlutterTaskQueue>* PlatformMessageHandlerIos::MakeBackgroundTaskQueue() {
```

Comment:
Similarly, do we need a `__attribute__((ns_returns_autoreleased))` for this method in case it is ever used from MRC?  Are we sure that it isn't?  Maybe we could add an arc assert in the header.

Without it I think ARC reserves the right to return a retained value, expecting the call site to release it.  A MRC client would expect it to be autoreleased.

Reply:
I thought ARC followed the standard Create/Copy naming rules.

Reply:
https://clang.llvm.org/docs/AutomaticReferenceCounting.html#conversion-to-retainable-object-pointer-type-of-expressions-with-known-semantics discusses what happens with [audited functions](https://clang.llvm.org/docs/AutomaticReferenceCounting.html#arc-misc-c-retainable-audit), which are new since I had last looked at this in detail. We could add the audit annotation to all of our helpers that follow standard semantics (which I hope is all of them, but we should check) as we go to make it clearer that we know what the semantics are.

So far I can't find discussion of what happens when functions *aren't* audited, but I'm almost positive that when ARC first rolled out the rule was that it would follow the naming rules (i.e., that everything acted audited). It seems unlikely they would have retroactively changed that since it would have been massively breaking, but if we mark as audited it'll be moot.

Reply:
Here's the hypothetical I was thinking of.  This is how the code will look for those that are using this function from MRC.

```c++
// MRC
void Foo::foo() {
  bar_ = [platform_message_handler_.MakeBackgroundTaskQueue() retain];
}
```

I checked the output for the following:
```c++
@interface Foo : NSObject
@end

@implementation Foo
@end

Foo* MakeFoo() {
  return [[Foo alloc] init];
}
```

The output is
```asm
__Z7MakeFoov:                           ; @_Z7MakeFoov
	.cfi_startproc
; %bb.0:
	stp	x29, x30, [sp, #-16]!           ; 16-byte Folded Spill
	.cfi_def_cfa_offset 16
	mov	x29, sp
	.cfi_def_cfa w29, 16
	.cfi_offset w30, -8
	.cfi_offset w29, -16
	adrp	x8, _OBJC_CLASSLIST_REFERENCES_$_@PAGE
	ldr	x0, [x8, _OBJC_CLASSLIST_REFERENCES_$_@PAGEOFF]
	bl	_objc_alloc_init
	ldp	x29, x30, [sp], #16             ; 16-byte Folded Reload
	b	_objc_autoreleaseReturnValue
	.cfi_endproc
                                        ; -- End function
	.globl	__Z3Barv                        ; -- Begin function _Z3Barv
	.p2align	2
```

So it looks like autorelease is the default behavior, `ns_returns_autoreleased` seems superfluous in this case.

Reply:
> So it looks like autorelease is the default behavior

That matches my recollection that ARC will honor CF-style naming conventions (since your method doesn't start with `Create` or `Copy`). But it's weird that I can't find anything that explicitly says that, and sprinkling `cf_audited_transfer` on our C functions as we go should be very easy.